src/QtImageReader.cpp

```diff
@@ -209,7 +209,7 @@ std::shared_ptr<Frame> QtImageReader::GetFrame(int64_t requested_frame)
 	}
 
 	// Scale image smaller (or use a previous scaled image)
-	if (!cached_image || (cached_image && cached_image->width() != max_width || cached_image->height() != max_height)) {
```

Comment:
If cached_image is NULL, will this still work? Or do we still need `cached_image && cached_image->...`

Reply:
In case if `cached_image` is NULL then the second part of the logical expression will not be evaluated.